Cabal/Distribution/Package.hs

```diff
@@ -83,15 +98,19 @@ unPackageName (PackageName s) = s
 -- Note: No validations are performed to ensure that the resulting
 -- 'PackageName' is valid
 mkPackageName :: String -> PackageName
+#if MIN_VERSION_bytestring(0,10,4)
+mkPackageName = PackageName . Data.String.fromString
```

Comment:
this is a hack for benchmarking which assumes that `String` is made up of the latin1 code points only; otherwise the construction is lossy